help_to_heat/portal/download_views.py

```diff
@@ -181,22 +183,16 @@ def download_feedback_view(request):
     return response
 
 
-@require_http_methods(["GET"])
-@decorators.requires_service_manager
-def download_referrals_all_view(request):
-    referrals = portal_models.Referral.objects.all().order_by("referral_id")
-    downloaded_at = timezone.now()
-    file_name = downloaded_at.strftime("%d-%m-%Y %H_%M")
-    response = create_referral_xlsx(referrals, file_name, exclude_pii=True)
-    return response
-
-
 @require_http_methods(["GET"])
 @decorators.requires_service_manager
 def download_referrals_last_week_view(request):
-    referrals = portal_models.Referral.objects.filter(created_at__gte=datetime.now() - timedelta(days=7)).order_by("referral_id")
-    downloaded_at = timezone.now()
-    file_name = downloaded_at.strftime("%d-%m-%Y %H_%M")
+    today = date.today()
+    end_of_last_week = today - timedelta(days=today.weekday())
+    start_of_last_week = end_of_last_week - timedelta(days=7)
+    referrals = portal_models.Referral.objects.filter(
+        created_at__gte=start_of_last_week, created_at__lt=end_of_last_week
+    ).order_by("referral_id")
+    file_name = f"Weekly Referrals ({datetime.strptime(str(start_of_last_week), '%Y-%m-%d').strftime('%d-%m-%Y')})"
```

Comment:
This is perhaps a little messy, but it seems there's no method for just formatting a date, so it seems we have to parse the date start_of_last_week as a string to a datetime, and then format this back into a string with just the date.

Reply:
Are you sure about this? I was able to just run:
```
start_of_last_week.strftime("%d-%m-%Y")
```

Here it is in the docs:
https://docs.python.org/3/library/datetime.html#datetime.date.strftime

> `date.strftime(format)`
Return a string representing the date, controlled by an explicit format string. Format codes referring to hours, minutes or seconds will see 0 values. See also [strftime() and strptime() Behavior](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior) and [date.isoformat()](https://docs.python.org/3/library/datetime.html#datetime.date.isoformat).

Reply:
Ah, thank you! It didn't seem quite right. I've changed this 😄